torch/nn/parallel/distributed.py

```diff
@@ -223,21 +223,26 @@ def __init__(self, module, device_ids=None,
         )
 
         self.is_multi_device_module = len({p.device for p in module.parameters()}) > 1
-        self.is_cuda = all([p.device.type == 'cuda' for p in module.parameters()])
-
-        if not self.is_cuda or self.is_multi_device_module:
+        distinct_device_type = {p.device.type for p in module.parameters()}
+        assert len(distinct_device_type) == 1, (
+            "DistributedDataParallel's input module must be on "
+            "the same device tpye, but input module parameters locate in {}."
```

Comment:
tpye -> type

Shall rephrase the first part of the sentence to "DistributedDataParallel's input module must be on the same type of devices, ..."